src/sentry/sentry_metrics/querying/api.py

```diff
@@ -121,98 +313,139 @@ def _generate_full_series(
+        _group(
+            execution_result.data,
+            lambda value, group: group.add_series_entry(
+                value.get("time"), value.get("aggregate_value")
+            ),
+        )
+
+        # We group the total data.
+        _group(
+            execution_result.totals,
+            lambda value, group: group.add_total(value.get("aggregate_value")),
+        )
+
+        meta = execution_result.meta
         for meta_item in meta:
-            meta_name = meta_item.get("name")
-            meta_type = meta_item.get("type")
+            meta_name = meta_item["name"]
+            meta_type = meta_item["type"]
 
-            # Since we have to handle multiple time series, we map the aggregate value to the actual
-            # metric name that was queried.
-            if meta_name == "aggregate_value":
-                intermediate_meta[query_result.name] = meta_type
-            else:
-                intermediate_meta[meta_name] = meta_type
+            # The meta of each query, contains the metadata for each field in the result. In this case, we want to map
+            # the aggregate value type to the actual query name, which is used from the outside to recognize the query.
+            name = execution_result.query_name if meta_name == "aggregate_value" else meta_name
+            intermediate_meta.append(QueryMeta(name=name, type=meta_type))
 
-    translated_groups = []
-    for group_key, group_metrics in sorted(intermediate_groups.items(), key=lambda v: v[0]):
-        # This case should never happen, since if we have start and intervals not None
-        assert start is not None and end is not None and intervals is not None
+    # If we don't have time bounds and an interval, we can't return anything.
+    assert start is not None and end is not None and interval is not None
 
-        start_seconds = int(start.timestamp())
-        num_intervals = len(intervals)
+    # We build the intervals that we will return to the API user.
+    intervals = _build_intervals(start, end, interval)
 
-        series = {}
-        totals = {}
+    translated_groups = []
+    for group_key, group_metrics in sorted(intermediate_groups.items(), key=lambda v: v[0]):
+        translated_serieses: Dict[str, Sequence[ResultValue]] = {}
+        translated_totals: Dict[str, ResultValue] = {}
         for metric_name, metric_values in sorted(group_metrics.items(), key=lambda v: v[0]):
-            series[metric_name] = _generate_full_series(
-                start_seconds, num_intervals, interval, metric_values[0]
+            series = metric_values.series
+            total = metric_values.total
+
+            # We generate the full series by passing as default value the identity of the totals, which is the default
+            # value applied in the timeseries.
+            translated_serieses[metric_name] = _generate_full_series(
+                int(start.timestamp()), len(intervals), interval, series, _get_identity(total)
```

Comment:
We have the total data type to determine the identify of a specific value.

Reply:
This is assuming that we always have the totals query, but since in the future this might change, we might have to find another way.